pkg/controller/machinehealthcheck/machinehealthcheck_controller.go

```diff
@@ -213,6 +221,24 @@ func remediate(r *ReconcileMachineHealthCheck, machine *mapiv1.Machine) (reconci
 
 		// apply remediation logic, if at least one condition last more than specified timeout
 		if unhealthyForTooLong(nodeCondition, conditionTimeout) {
+			if isBareMetalProvider(infra) {
+				// we already have reboot annotation on the node, stop reconcile
+				if _, ok := node.Annotations[annotationReboot]; ok {
```

Comment:
Are we sure we want to annotate node and not the machine?

Reply:
@vikaschoudhary16 To be honest, I do not have a strong opinion, but in node case, if reboot failed and we delete the node I do not need in addition to remove the annotation, so currently I prefer this way

Reply:
@cynepco3hahue  Lets say 'node' is k8s node object and 'instance' is the corresponding machine. Assuming you meant, if an instance reboot fails and we delete the instance, there will be no need to remove annotation from the k8s node object. Did i get you right?

If yes, i am not sure if external controller should delete the instance on reboot failure. Because instance deletion will not remediate the machine actually. New instance will not be able to become k8s node because of failing CSR. Instead, on reboot failing, if machine object is deleted, that would remediate the situation.

Another thing which i am not very confident is that if controller handling "reboot" annotation, should do "delete". Will not it be a bit inconsistent?

Reply:
@vikaschoudhary16 Nope I meant the k8s node, under the bare metal platform we do not want to delete the machine object at all(it the reason why we have all this W/A and additional controllers), but we still want to release workload in case when reboot failed, so we will delete the node(you can do it because you already know that you rebooted the host via power management and it does not run any workloads)

Reply:
@cynepco3hahue copying from the slack for the record:
ok, so if node is annotated and to handle this first, node is deleted and then instance is rebooted. In failure case, machine object will hang there without NodeRef. And an alert will generated after some configured time ~10 min.
Now if we discuss, alternative way, in which machine is being annotated..when alert for nil NodeRef, would this annotation of node reboot be helpful for the operator to understand why NodeRef is nil? As a clue on machine object

Reply:
/cc @enxebre